### PR TITLE
Introduce navbar-primary class rather than overriding Bootstrap navbar

### PR DESCRIPTION
**Pull request by jgolieb**
Per our team decision, I have moved all the Fluid-style navbar customizations behind a new modifier class, `navbar-primary`. As a result, I was also able to get rid of the custom `_header.scss` and `_footer.scss` components and just use `<nav>` tags for those elements on the demo page.

### CODE REVIEW

File: scss/components/_navbar.scss [hunk 1]
```diff
@@ -6,7 +6,29 @@
  * of all or any part of the contents of this file is strictly prohibited.
  */
 
-.navbar {
+body {
+
+  > .navbar {
```
Comment (mihai-mihalache) [inline scss/components/_navbar.scss]: you can not assume that a navbar will always be a direct descendant of the body thus raising specificity this way might never create the desired effect

Comment (jgolieb) [reply scss/components/_navbar.scss]: The point of this is to target only navbars that are the direct descendant of the body. These styles are for creating the top horizontal navbar that is intended to contain the global breadcrumbs and such. It must be a direct descendant of the body (and be a sibling of the .navbar-primary) for the Fluid navbar collapse/expand to work properly. This is all part of the recommended/assumed layout that Fluid apps should use, and our demo page is providing an example of it. (It also affects the global nav-footer navbar.)

Comment (mihai-mihalache) [reply scss/components/_navbar.scss]: What if you can't get it as a direct descendant of the body? There are some frameworks that force a wrapper around the whole application

Comment (jgolieb) [reply scss/components/_navbar.scss]: OK, I'll have to add another custom class for the header and footer then.

**Review by mihai-mihalache — APPROVED** [review]

**Pull request merged**